python/task_manager/worker_thread.py

```diff
@@ -57,6 +55,7 @@ def shut_down(self):
         """
         Shut down the thread and wait for it to exit before returning
         """
+        self._results_poller = None
```

Comment:
minor: is this a GC thing or is it a way to immediately 'mute' the worker once shut down has been called to ensure no 'straggling tasks' emit signals? May be worth a comment for future readers.